test/tests/functional/pbs_reservations.py

```diff
@@ -883,3 +883,80 @@ def test_fail_confirm_resv_message(self):
         # The scheduler should log reason why it was denied
         self.scheduler.log_match(rid + ";PBS Failed to confirm resv: " +
                                  "Insufficient amount of resource: ncpus")
+
+    def test_advance_resv_with_multinode_job_array(self):
+        """
+        Test multinode job array with advance reservation
+        """
+        if (len(self.moms) < 2):
+            self.skip_test("Test requires 2 moms as input")
+        a = {'resources_available.ncpus': 4}
+        for mom in self.moms.values():
+            self.server.manager(MGR_CMD_SET, NODE, a, id=mom.shortname)
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'job_history_enable': 'True'},
+                            id=self.server.shortname)
+        now = int(time.time())
+        a = {'Resource_List.select': '2:ncpus=2',
+             'Resource_List.place': 'scatter',
+             'reserve_start': now + 30,
+             'reserve_end': now + 300}
+        r = Reservation(PBSROOT_USER, attrs=a)
+        rid = self.server.submit(r)
+        exp_attr = {'reserve_state': (MATCH_RE, "RESV_CONFIRMED|2")}
+        self.server.expect(RESV, exp_attr, id=rid)
+        resv_queue = rid.split(".")[0]
+
+        attrs = {ATTR_q: resv_queue, ATTR_J: '1-5',
+                 'Resource_List.select': '2:ncpus=1'}
+        j = Job(PBSROOT_USER)
+        j.set_sleep_time(60)
+        j.set_attributes(attrs)
+        jid = self.server.submit(j)
+        subjid = []
+        for i in range(1, 6):
+            subjid.append(j.create_subjob_id(jid, i))
+
+        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
+        self.server.expect(RESV, a, id=rid, offset=30)
+        self.server.expect(JOB, {'job_state': 'B'}, id=jid)
+        self.server.expect(JOB, {'job_state=R': 2}, count=True,
+                           extend='t', id=jid)
+        self.server.expect(JOB, {'job_state=Q': 3}, count=True,
+                           extend='t', id=jid)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=subjid[3])
+        self.server.deljob(subjid[3])
```

Comment:
I think self.server.delete() is the right PTL interface to use. (and other places in the test plan)